Add multi delete and Changed addressbook.json to GMEdata.json

lyueyang opened the pull request.

lyueyang committed:
Multiple delete feature added

lyueyang committed:
Change name of addressbook.json

Changed to GMEdata.json

lyueyang committed:
Adjusted invalid index message reply

lyueyang committed:
Merge branch 'master' into add-multi-delete

lyueyang committed:
Altered .gitignore to stop tracking preferences.json

UserPrefs is where the name for addressbook.json is defined instead

lyueyang committed:
Rectify whitespace issues

Whitespace issues highlighted by run-checks.sh

JoelHo commented on src/main/java/seedu/address/commons/core/Messages.java:
```diff
@@ -7,7 +7,8 @@
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PASSENGER_DISPLAYED_INDEX = "The passenger index provided is invalid";
+    public static final String MESSAGE_INVALID_PASSENGER_DISPLAYED_INDEX =
+            "One of the passenger index provided is invalid";
```
Should this be indices/indexes?

lyueyang replied:
Rectified this, thanks for pointing it out!

JoelHo commented on src/main/java/seedu/address/logic/commands/DeleteCommand.java:
```diff
@@ -22,36 +23,44 @@
-        if (!model.deletePassenger(passengerToDelete)) {
-            throw new CommandException(String.format(MESSAGE_DELETE_PASSENGER_FAIL_HAS_POOL, passengerToDelete));
+        for (Passenger p : targetedPassengers) {
+            if (!model.deletePassenger(p)) {
+                throw new CommandException(String.format(MESSAGE_DELETE_PASSENGER_FAIL_HAS_POOL, p));
+            }
         }
-        return new CommandResult(String.format(MESSAGE_DELETE_PASSENGER_SUCCESS, passengerToDelete));
+
+        return new CommandResult(String.format(MESSAGE_DELETE_PASSENGER_SUCCESS, targetedPassengers.size()));
```
Should we implement it like this (i.e. delete the passengers at index until we hit one with a pool, ignore the rest)? Perhaps a better implementation is to delete all that don't have a pool then error at the end. Also, I think it would be more intuitive if we printed the deleted passengers names rather than size.

lyueyang replied:
Resolved this by changing logic in implementation as discussed

JoelHo commented on src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:
```diff
@@ -18,8 +21,17 @@
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
+            String[] arguments = args.split("\\s+");
+            List<Index> indexes = new ArrayList<>();
+
+            for (String s : arguments) {
+                if (s.length() > 0) {
+                    Index index = ParserUtil.parseIndex(s);
+                    indexes.add(index);
+                }
+            }
+
+            return new DeleteCommand(indexes);
```
Would it be better to use argmultimap here rather than implement a custom parser?

lyueyang replied:
Have moved this to ParserUtil as discussed

JoelHo reviewed (changes requested):
Some code quality issues that needs feedback for.

lyueyang committed:
Modify error messages

Error messages print passenger names to improve usability

lyueyang committed:
Fix bugs related to printed messages

Message for DeleteCommand was printing the same passenger for scenario where some passengers were deleted but others not as they are in pool.

lyueyang committed:
Refactor parsing command

Moved parsing to ParserUtil to neaten up DeleteCommandParser

lyueyang committed:
Merge branch 'add-multi-delete'

lyueyang committed:
Refactor parsing command

Parsing code moved from DeleteCommandParser to ParserUtil

lyueyang committed:
Merge branch 'master' into add-multi-delete

lyueyang committed:
Modify UG to reflect multi-delete

Also specified that users with a pool arrangement cannot be deleted.

JoelHo reviewed (approved):
LGTM
